magento/magento2
# Set process title when running cron job

### Description
When managing a busy server, it's helpful to understand what each process is doing. Currently there is no visibility for a systems administrator to see what task the `bin/magento cron:run` process is currently performing. The most information that's available is that some job (or set of jobs) is running within a PHP process that was started via the system task scheduler (read: cron daemon). Because of the way that Magento spawns additional processes for some groups (depending on their configuration) it is possible to sometimes narrow down the list of possible tasks because the group name is listed in the command arguments.

This pull request sets the title of the currently-running process to clearly list the group and job names.

<details><summary>Before screen-shot</summary>

![Screenshot_2021-10-04_22-59-50](https://user-images.githubusercontent.com/334786/135930916-0325b3a5-609b-4831-a534-f74fca30298a.png)

</details><details><summary>After screen-shot</summary>

![Screenshot_2021-10-04_22-56-17](https://user-images.githubusercontent.com/334786/135930903-022cf1d7-87a5-48ee-9250-83bd1bb7576d.png)

</details>

### Related Pull Requests
*None*


### Manual testing scenarios
1. Run the standard Magento cron with `bin/magento cron:run` when several long-running tasks are scheduled
1. Observe lack of (or presence of after this change) information about which job is running in the process tree (using tools like `ps` or `htop` on Unix - untested on Windows, but the functionality should work there too)

### Questions or comments
Should the path to Magento be listed in the process title too? If there are multiple Magento instances on a single server, they should be running under different users. I can't think of a sensible non-development environment where multiple Magento instances run under the same system user.

### Contribution checklist


### Resolved issues:
1. [x] resolves magento/magento2#34321: Set process title when running cron job
